server/cla.go

```diff
@@ -54,80 +59,108 @@ func (s *Server) checkCLA(pr *model.PullRequest) {
 		return
 	}
 
+	body, errCSV := s.getCSV()
+	if errCSV != nil {
+		return
+	}
+
+	if !isNameInCLAList(strings.Split(string(body), "\n"), username) {
+		comments, _, err := s.GithubClient.Issues.ListComments(context.TODO(), pr.RepoOwner, pr.RepoName, pr.Number, nil)
+		if err != nil {
+			mlog.Error("pr_error", mlog.Err(err))
+			return
+		}
+		_, found := findNeedsToSignCLAComment(comments, s.Config.Username)
+		if !found {
+			s.sendGitHubComment(pr.RepoOwner, pr.RepoName, pr.Number, strings.Replace(s.Config.NeedsToSignCLAMessage, "USERNAME", "@"+username, 1))
+		}
+		status := &github.RepoStatus{
+			State:       github.String(stateError),
+			Description: github.String(fmt.Sprintf("%v needs to sign the CLA", username)),
+			TargetURL:   github.String(s.Config.SignedCLAURL),
+			Context:     github.String(s.Config.CLAGithubStatusContext),
+		}
+		mlog.Debug("will post error on CLA", mlog.String("user", username))
+		_, _, errStatus := s.GithubClient.Repositories.CreateStatus(context.TODO(), pr.RepoOwner, pr.RepoName, pr.Sha, status)
```

Comment:
Does it make sense to use `createRepoStatus` here?

Reply:
It's definitely shorter, I will use that.

Reply:
Are you still planing to do that?

Reply:
Done